sqlalchemy_dremio/base.py

```diff
@@ -0,0 +1,266 @@
+
+    function_rewrites = {'current_date': 'now',
+                          'current_timestamp': 'now',
+                          'length': 'len',
+                          }
+    def visit_function(self, func, **kwargs):
+        """Dremio function names differ from the ANSI SQL names;
+        rewrite common ones"""
+        func.name = self.function_rewrites.get(func.name, func.name)
+        return super(DremioCompiler, self).visit_function(func)
+
+    def for_update_clause(self, select):
+        """FOR UPDATE is not supported by Dremio; silently ignore"""
+        return ''
+
+    # Strip schema
+    def visit_table(self, table, asfrom=False, **kwargs):
+        if asfrom:
+            return self.preparer.quote(table.name, table.quote)
+        else:
+            return ""
+
+    def visit_join(self, join, asfrom=False, **kwargs):
+        return ('(' + self.process(join.left, asfrom=True) + \
+                (join.isouter and " LEFT OUTER JOIN " or " INNER JOIN ") + \
+                self.process(join.right, asfrom=True) + " ON " + \
+                self.process(join.onclause) + ')')
+
+    def visit_extract(self, extract, **kw):
+        field = self.extract_map.get(extract.field, extract.field)
+        return 'DATEPART("%s", %s)' % \
+                    (field, self.process(extract.expr, **kw))
+
+class DremioDDLCompiler(compiler.DDLCompiler):
+    def get_column_specification(self, column, **kwargs):
+        if column.table is None:
+            raise exc.CompileError(
+                            "dremio requires Table-bound columns "
+                            "in order to generate DDL")
+
+        colspec = self.preparer.format_column(column)
+        seq_col = column.table._autoincrement_column
+        if seq_col is column:
+            colspec += " AUTOINCREMENT"
+        else:
+            colspec += " " + self.dialect.type_compiler.process(column.type)
+
+            if column.nullable is not None and not column.primary_key:
+                if not column.nullable or column.primary_key:
+                    colspec += " NOT NULL"
+                else:
+                    colspec += " NULL"
+
+            default = self.get_column_default_string(column)
+            if default is not None:
+                colspec += " DEFAULT " + default
+
+        return colspec
+
+    def visit_drop_index(self, drop):
+        index = drop.element
+        self.append("\nDROP INDEX [%s].[%s]" % \
+                        (index.table.name,
+                        self._index_identifier(index.name)))
+
+class DremioIdentifierPreparer(compiler.IdentifierPreparer):
+    reserved_words = compiler.RESERVED_WORDS.copy()
+    reserved_words.update(['value', 'text'])
+    def __init__(self, dialect):
+        super(DremioIdentifierPreparer, self).\
+                __init__(dialect, initial_quote='[', final_quote=']')
+
+
+
+class DremioDialect(default.DefaultDialect):
+    colspecs = {
```

Comment:
look here for better types https://github.com/JohnOmernik/sqlalchemy-drill/blob/d213e26cd6b5ae07ad675e72d015ea4d7d03773b/sqlalchemy_drill/sadrill.py#L42